lib/natalie/compiler/backends/cpp_backend.rb

```diff
@@ -80,7 +80,11 @@ def symbols_var_name
 
       def obj_files
         rb_files = Dir.children(File.expand_path('../../../../src', __dir__)).grep(/\.rb$/)
-        list = rb_files.sort.map { |name| name.split('.').first }
+        list = rb_files.sort.map { |name|
+          ns = name.split('.')
+          raise "Found broken src/*.rb file #{name.inspect}" if ns.first.empty?
```

Comment:
I've reverted this commit (and another commit causing test issues) but will throw another issue for discussion, because there is probably a better way to do it than this.

Reply:
Linking issue #1034 for posterity